crates/broker/src/order_picker.rs

```diff
@@ -524,19 +549,45 @@ where
 
         for (order_id, order) in order_res {
             let picker_clone = self.clone();
-            tasks.spawn(async move {
-                match picker_clone.price_order(order_id, &order).await {
-                    Ok(true) => Ok(Some(order_id)),
-                    Ok(false) => Ok(None),
-                    Err(err) => Err((order_id, err)),
-                }
-            });
+            tasks.spawn(
+                async move { picker_clone.price_order_and_update_db(order_id, &order).await },
+            );
         }
 
         Ok(())
     }
 }
 
+/// The capacity of the order picker, if there is a limit on the number of concurrent locks.
+#[derive(Debug, PartialEq)]
+enum Capacity {
+    /// There are no pending or currently locked orders.
+    Idle(u32),
+    /// There are orders that are picked to be locked but not fulfilled yet.
+    PartiallyLocked(u32),
+    /// There is no concurrent lock limit.
+    Unlimited,
+}
+
+impl Capacity {
+    fn request_size(&self, pricing_tasks: usize) -> u32 {
+        match self {
+            Capacity::Idle(capacity) | Capacity::PartiallyLocked(capacity) => std::cmp::min(
+                capacity.saturating_sub(u32::try_from(pricing_tasks).expect("tasks u32 overflow")),
+                10,
```

Comment:
It's not clear from reading this why there is a cap of 10. My understanding is that this limits the number of concurrent pricing jobs are running at any time, such that if the max pending locks is more than 10 it will still only work on pricing 10 at a time. Is that accurate? If so, it would be good to add a comment to that effect and possible extract this constant to a named value.

Reply:
[`d43c7e1` (#433)](https://github.com/boundless-xyz/boundless/pull/433/commits/d43c7e10ca821801266cb01e92ffcda14b3d3718)

Yes, that is accurate. Previously this was limited by the fact that orders were pulled with a sleep inbetween, and this just keeps the behaviour similar after switching to doing in batches.